Apps/Database/src/Delegates/DbProfileDelegate.cs

```diff
@@ -299,6 +299,23 @@ public int GetRecurrentUserCount(int dayCount, DateTime startDate, DateTime endD
             return recurrentCount;
         }
 
+        /// <inheritdoc/>
+        public IDictionary<string, int> GetLastLoginClientCounts(DateTime startDate, DateTime endDate)
+        {
+            Dictionary<string, int> loginClientCounts = this.dbContext.UserProfile
+                .Select(x => new { x.HdId, x.LastLoginClientCode, x.LastLoginDateTime })
+                .Concat(
+                    this.dbContext.UserProfileHistory.Select(x => new { x.HdId, x.LastLoginClientCode, x.LastLoginDateTime }))
+                .Where(x => x.LastLoginClientCode != null && x.LastLoginDateTime >= startDate && x.LastLoginDateTime <= endDate)
+                .Select(x => new { x.HdId, x.LastLoginClientCode, lastLoginDate = GatewayDbContext.DateTrunc("days", x.LastLoginDateTime) })
+                .Distinct()
```

Comment:
Since the acceptance criteria is to return all logins in the date range with the given LastLoginClientCode, distinct on HDID, we should remove lastLoginDate from the Select().

Reply:
@ninosamson before approving we need to discuss this.